src/http/FileUploadModule.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "http/FileUploadModule.h"
+
+// System headers
+#include <memory>
+
+// Third-party headers
+#include <httplib.h>
+
+// Qserv headers
+#include "http/Exceptions.h"
+#include "http/RequestQuery.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace lsst::qserv::http {
+
+FileUploadModule::FileUploadModule(string const& authKey, string const& adminAuthKey,
+                                   httplib::Request const& req, httplib::Response& resp,
+                                   httplib::ContentReader const& contentReader)
+        : BaseModule(authKey, adminAuthKey), _req(req), _resp(resp), _contentReader(contentReader) {}
+
+void FileUploadModule::execute(string const& subModuleName, http::AuthType const authType) {
+    _subModuleName = subModuleName;
+    try {
+        if (!_req.is_multipart_form_data()) {
+            throw AuthError(context() + "the request is not a multipart form data");
+        }
+        unique_ptr<httplib::MultipartFormData> currentFile;
+        auto const processEndOfEntry = [&]() {
+            if (currentFile != nullptr) {
+                if (!currentFile->filename.empty()) {
+                    onEndOfFile();
+                } else {
+                    body().objJson[currentFile->name] = currentFile->content;
+                }
+            }
+        };
+        _contentReader(
+                [&](httplib::MultipartFormData const& file) -> bool {
```

Comment:
Do you really want `[&]` here?  Arg is already a const ref...

Reply:
The problem is that this lambda calls 3 methods of the current class. So, I would need at least ``this`` to make the compiler happy. It also modifies a value of ``currentFile``. It's easier to pass ``&``.